s3_to_redshift.go

```diff
@@ -88,6 +107,14 @@ func runCopy(db *redshift.Redshift, inputConf s3filepath.S3File, inputTable reds
 func main() {
 	flag.Parse()
 
+	// verify that timeGranularity is a supported value. for convenience,
+	// we use the convention that granularities must be valid PostgreSQL dateparts
+	// (see: http://www.postgresql.org/docs/8.1/static/functions-datetime.html#FUNCTIONS-DATETIME-TRUNC)
+	supportedGranularities := map[string]bool{"hour": true, "day": true}
+	if !supportedGranularities[*timeGranularity] {
+		panic(fmt.Sprintf("Unsupported granularity, must be one of %v", reflect.ValueOf(supportedGranularities).MapKeys()))
```

Comment:
I'd strongly prefer we don't introduce `reflect` as a dependency, particularly when we only use it for getting keys of a map.